packages/utils/src/formatCurrencyAmount.ts

```diff
@@ -0,0 +1,12 @@
+export const formatCurrencyAmount = (amount: number, locale = 'en') => {
+    if (
+        typeof amount !== 'number' ||
+        Number.isNaN(amount) ||
+        !Number.isFinite(amount) ||
+        amount > Number.MAX_SAFE_INTEGER
+    ) {
+        return;
+    }
+
+    return Intl.NumberFormat(locale, { style: 'decimal' }).format(amount);
```

Comment:
If the number formatting for satoshis is supposed to be a part of this issue, it has to be mentioned in the issue description. cc @hynek-jina

Reply:
Expected rounding has to be defined too. `Intl.NumberFormat().format()` has default `maximumFractionDigits: 3` so `Intl.NumberFormat('en', { style: 'decimal' }).format('1.9999')` returns `2`.  

`style: 'decimal'` is also the default value, btw.

Reply:
And the next step would [be adding locale](https://github.com/trezor/trezor-suite/issues/5545) :D

Reply:
We can't threat crypto amounts the same way as fiat amounts. Fiat amounts are mostly only informative in Suite, but crypto amounts has to be always precise and error-proof so it has to be consistent in Suite with what is on trezor display.

Reply:
Hmm, `sats` cannot have decimals at all though and currently the util is only used there. So there can be a default value of 0 and and argument controlling `minimumFractionDigits`?

Reply:
I would suggest to 
- add `maximumFractionDigits: 20` (max allowed value) to prevent rounding,
- add test cases for decimal numbers because it is a util that somebody could reuse later and it's not obvious how it behaves with decimal number without the tests,
- move it out of @trezor/utils package (to suite utils) see [comment](https://github.com/trezor/trezor-suite/pull/5810#discussion_r925557627).

Reply:
done